### PR TITLE
Some fixes to linux builds

### PR DESCRIPTION
**Pull request by AiraYumi**
We plan to make some fixes in this RP.

- [x] compile unuse flag remove from linux build
- [x] use fltk 3p library
- [x] Revert #1107
- [x] Verify that you can build with tracy enabled.
- [x] Build Test

This is currently a work in progress and the PR may be discarded, so it will be drafted.

### CODE REVIEW

**Comment by AiraYumi** [comment]
@Nicky-D / @marchcat / @bennettgoble / @nat-goodspeed
Do I need to enable tracy in build_linux.yaml? for testing.

File: indra/cmake/CEFPlugin.cmake
Comment (Nicky-D) [inline indra/cmake/CEFPlugin.cmake:1]: This is not enough to enable CEF. 
Also  CEF_LIBRARY is not defined for Linux

CEF will be coming with this branch: https://github.com/Nicky-D/viewer/tree/feature/linux_media

File: autobuild.xml
Comment (Nicky-D) [inline autobuild.xml:1]: Both of this changes will be coming via https://github.com/Nicky-D/viewer/tree/feature/linux_media

File: indra/cmake/CEFPlugin.cmake
Comment (AiraYumi) [reply indra/cmake/CEFPlugin.cmake:1]: Is it better to rollback changes regarding CEF?

Comment (Nicky-D) [reply indra/cmake/CEFPlugin.cmake:1]: Yes please. This will be easier for the Linden merging, as there will be less conflicts

File: autobuild.xml
Comment (Nicky-D) [inline autobuild.xml:873]: Thank you for removing googlemock. That one has been bothering me for ages :)

**Comment by AiraYumi** [comment]
Removed CEF updates to reduce contention.

**Comment by AiraYumi** [comment]
remove change for cef.

File: indra/cmake/UI.cmake
Comment (Nicky-D) [inline indra/cmake/UI.cmake:20]: Thank you for adding that. After this, it _should_ be possible to drop llibfltk1.3-dev from .github/workflows/build_linux.yaml

**Comment by AiraYumi** [comment]
After #1146 is merged, pull the changes into this PR.

**Comment by AiraYumi** [comment]
We will move from Draft to a normal pull request, but if we want to merge this PR, we would like to do it after incorporating the changes made after #1146 was merged.

**Comment by bennettgoble** [comment]
Hi, @AiraYumi. Thank you for the PR. Could you please remove any work that has already been merged into another Linden branch (example: https://github.com/secondlife/viewer/pull/1054 etc.)? We will take care of merging work between release vehicles and don't need it to be re-applied.

**Review by bennettgoble — CHANGES_REQUESTED** [review]
PR needs already-merged work removed.

**Comment by AiraYumi** [comment]
Removed existing merge.
Once you have updated to the latest expat build and confirmed that it can be built, there is no problem with merging.

**Comment by AiraYumi** [comment]
3p-libraries bump complete.

build testing.

File: indra/cmake/00-Common.cmake
Comment (Nicky-D) [inline indra/cmake/00-Common.cmake:201]: Where did you encounter this issue. I have not been able to reproduce with
`gcc version 13.2.1 20240210 (Gentoo 13.2.1_p20240210 p14)`

Comment (AiraYumi) [reply indra/cmake/00-Common.cmake:201]: The problem occurs when bumping boost to 1.84.
It does not occur now and can be removed, but it will occur if you update to 1.84 in the future, so you will need to fix it anyway.

Comment (Nicky-D) [reply indra/cmake/00-Common.cmake:201]: Got it. We shall look at it again when then time comes. Maybe there's another way to get rid of the warning without having to disable it on the whole code base

**Comment by AiraYumi** [comment]
The build completed successfully in my local repository.
It worked fine with gcc13, so it's probably fine with gcc12 as well.
If there is no problem, please merge.

**Review by marchcat — APPROVED** [review]
LGTM, 
Thank you!

File: autobuild.xml
Comment (bennettgoble) [inline autobuild.xml:621]: It looks like you're updating expat, but only on linux. We generally try to update all platforms at once, and I have been working on that here: https://github.com/secondlife/viewer/issues/1078 Could you remove the autobuild.xml changes that are being worked on under #1078?

**Review by bennettgoble — APPROVED** [review]

**Comment by AiraYumi** [comment]
I removed the expat update.

**Pull request merged**
